File: v0dot2/project_environment_all/py_unittest/test_update_bombs.py
import unittest
from environment_all import GenericWorld

class TestGenericWorld(unittest.TestCase):

    def test_update_bombs(self):
        # Arrange
        world = GenericWorld()
        world.bombs = [
            Bomb(1, 1, 3),
            Bomb(2, 2, 2),
            Bomb(3, 3, 1)
        ]

        # Act
        world.update_bombs()

        # Assert
        self.assertEqual(len(world.explosions), 2)
        self.assertEqual(world.explosions[0].coords, [(1, 1), (2, 2), (3, 3)])
        self.assertEqual(world.explosions[1].coords, [(2, 2), (3, 3)])
        self.assertEqual(world.explosions[0].screen_coords, [(21, 21), (22, 22), (23, 23)])
        self.assertEqual(world.explosions[1].screen_coords, [(22, 22), (23, 23)])
        self.assertEqual(world.explosions[0].owner, None)
        self.assertEqual(world.explosions[1].owner, None)
        self.assertFalse(world.bombs[0].active)
        self.assertFalse(world.bombs[1].active)
        self.assertFalse(world.bombs[2].active)


if __name__ == '__main__':
    unittest.main()

File: previous_versions/v0dot3/project_environment_all/js_unittest/test_update_bombs.js
const { TestGenericWorld } = require('./path/to/your/testFile');
const { GenericWorld } = require('./path/to/your/GenericWorld');
const { Bomb } = require('./path/to/your/Bomb');

describe('TestGenericWorld', () => {
  test('test_update_bombs', () => {
    // Arrange
    const world = new GenericWorld();
    world.bombs = [
      new Bomb(1, 1, 3),
      new Bomb(2, 2, 2),
      new Bomb(3, 3, 1)
    ];

    // Act
    world.update_bombs();

    // Assert
    expect(world.explosions.length).toBe(2);
    expect(world.explosions[0].coords).toEqual([[1, 1], [2, 2], [3, 3]]);
    expect(world.explosions[1].coords).toEqual([[2, 2], [3, 3]]);
    expect(world.explosions[0].screen_coords).toEqual([[21, 21], [22, 22], [23, 23]]);
    expect(world.explosions[1].screen_coords).toEqual([[22, 22], [23, 23]]);
    expect(world.explosions[0].owner).toBeNull();
    expect(world.explosions[1].owner).toBeNull();
    expect(world.bombs[0].active).toBeFalsy();
    expect(world.bombs[1].active).toBeFalsy();
    expect(world.bombs[2].active).toBeFalsy();
  });
});